src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelBindingHelper.cs

```diff
@@ -100,7 +100,20 @@ internal static void AddModelErrorBasedOnExceptionType(ModelBindingContext bindi
         {
             if (IsFormatException(ex))
             {
-                bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                var modelState = bindingContext.ModelState[bindingContext.ModelName];
+                if (modelState == null)
+                {
+                    bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                }
+                else
+                {
+                    // Convert FormatExceptions to Invalid value messages.
+                    var errorMessage = Resources.FormatModelBinderUtil_ValueInvalid(
+                        modelState.Value.AttemptedValue,
+                        bindingContext.ModelName);
+
+                    bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, errorMessage);
+                }
```

Comment:
should we just be doing this inside of modelstatedictionary? then it happens for all model errors no matter what.